.github/workflows/shared-modules.yml

```
@@ -0,0 +1,98 @@
+# This workflow is responsible for testing shared modules.
+name: Test Shared Modules Workflow
+
+on:
+  # This line enables manual triggering of this workflow.
+  workflow_dispatch:
+  # A workflow can be called by another workflow
+  workflow_call:
+    inputs:
+      pr:
+        description: "This is the PR number in case the workflow is being called in a pull request"
+        required: false
+        type: number
+
+  pull_request:
+    branches: [release, master]
+    paths:
+      - "app/shared/**"
+
+# Change the working directory for all the jobs in this workflow
+defaults:
+  run:
+    working-directory: app/shared
+
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    # Only run this workflow for internally triggered events
+    if: github.event_name == 'repository_dispatch' &&
+      github.event.client_payload.slash_command.sha != '' &&
+      contains(github.event.client_payload.pull_request.head.sha, github.event.client_payload.slash_command.sha)
+    defaults:
+      run:
+        working-directory: app/shared
+        shell: bash
+    
+    steps:
+
+      # Check out merge commit with the base branch in case this workflow is invoked via pull request
+      - name: Checkout the merged commit from PR and base branch
+        if: inputs.pr != 0
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: refs/pull/${{ inputs.pr }}/merge
+
+      # Checkout the code in the current branch in case the workflow is called because of a branch push event
+      - name: Checkout the head commit of the branch
+        if: inputs.pr == 0
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      
+      # Logging some params
+      - name: Figure out the PR number
+        run: echo ${{ inputs.pr }}
+
+      - name: Print the Github event
+        run: echo ${{ github.event_name }}
+
+      # Depot CLI helps building Docker images via Depot's remote builder infrastructure, faster.
+      - name: Set up Depot CLI
+        uses: depot/setup-action@v1
+      
+      # Timestamp will be used to create cache key
+      - id: timestamp
+        run: echo "::set-output name=timestamp::$(timestamp +'%Y-%m-%dT%H:%M:%S')"
+
+      # In case this is second attempt try restoring status of the prior attempt from cache
+      - name: Restore the previous run result
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/run_result
+          key: ${{ github.run_id }}-${{ github.job }}-${{ steps.timestamp.outputs.timestamp }}
```

```suggestion
          key: ${{ github.run_id }}-${{ github.job }}-shared-module
```
Instead of timestamp, we should be appending the github run & job id with the name of the service that is using the cache. In this case, it's `shared-module`.

---

Updated the key name for the step `Restore the previous run result`.